guard/src/commands/validate/common.rs

```diff
@@ -0,0 +1,262 @@
+                    continue;
+                },
+
+                _ => failed.extend(find_all_failing_clauses(each))
+            }
+        }
+    }
+    failed
+}
+
+pub(super) fn print_name_info<R, U, B>(
+    writer: &mut dyn Write,
+    info: &[NameInfo<'_>],
+    longest_rule_len: usize,
+    rules_file_name: &str,
+    data_file_name: &str,
+    retrieval_error: R,
+    unary_message: U,
+    binary_message: B) -> crate::rules::Result<()>
+    where R: Fn(&str, &str, &NameInfo<'_>) -> crate::rules::Result<String>,
+          U: Fn(&str, &str, &str, &NameInfo<'_>) -> crate::rules::Result<String>,
+          B: Fn(&str, &str, &str, &NameInfo<'_>) -> crate::rules::Result<String>
+{
+    for each in info {
+        let (cmp, not) = match &each.comparison {
+            Some(cmp) => (Some(cmp.operator), cmp.not),
+            None => (None, false)
+        };
+        // CFN = Resource [<name>] was not compliant with [<rule-name>] for property [<path>] because provided value [<value>] did not match expected value [<value>]. Error Message [<msg>]
+        // General = Violation of [<rule-name>] for property [<path>] because provided value [<value>] did not match expected value [<value>]. Error Message [<msg>]
+        // EQUALS failed at property path Properties.Encrypted because provided value [false] did not match with expected value [true].
+        match cmp {
+            None => {
+                // Block Clause retrieval error
+                writeln!(writer, "{}", retrieval_error(rules_file_name, data_file_name, each)?)?;
+            },
+
+            Some(cmp) => {
+                if cmp.is_unary() {
+                    writeln!(writer, "{}",
+                         unary_message(
+                             rules_file_name,
+                             data_file_name,
+                             match cmp {
+                                 CmpOperator::Exists => if !not { "did not exist" } else { "existed" },
+                                 CmpOperator::Empty => if !not { "was not empty"} else { "was empty" },
+                                 CmpOperator::IsList => if !not { "was not a list " } else { "was list" },
+                                 CmpOperator::IsMap => if !not { "was not a struct" } else { "was struct" },
+                                 CmpOperator::IsString => if !not { "was not a string " } else { "was string" },
+                                 _ => unreachable!()
+                             },
+                             each)?,
+                    )?;
+
+                }
+                else {
+                    // EQUALS failed at property path Properties.Encrypted because provided value [false] did not match with expected value [true].
+                    writeln!(writer, "{}",
+                        binary_message(
+                            rules_file_name,
+                            data_file_name,
+                            if not { "did" } else { "did not" },
```

Comment:
This is rather confusing. If not then did, else did not? Can we use variable names that go with the flow?
It would have been easier to read had it been

```
if not
 {"did not"}
else
 {"did"}
```

Reply:
@gandhek, `not` here is a boolean variable.

Reply:
`not` means exactly that, NOT or negation of the operator.  `!=, NOT EXISTS, NOT EMPTY`. What alternative names would you suggest?

Reply:
May be `not` can be named `negation_present`?